Static/DataSource.swift

```diff
@@ -101,8 +101,8 @@ public class DataSource: NSObject {
         refresh()
     }
 
-    private func refresh() {
-        refreshTableSections()
+    private func refresh(oldSections: [Section] = []) {
+        refreshTableSections(oldSections: oldSections)
```

Comment:
Hey guys, just FYI I'm testing this PR on my end in a project and these changes have surfaced a small bug related to dequeuing cells that have yet to be registered in the table view. If we just swap these two lines the issue is resolved.

By calling `refreshRegisteredCells()` first, we ensure any new cells that have been added to the sections are registered before updating the layout.